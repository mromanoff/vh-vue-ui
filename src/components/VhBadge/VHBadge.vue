<template>
  <div class="VHBadgeContainer">
    <span>
      <slot />
    </span>
    <span
      :class="computedClasses"
      class="VHBadge"
    >
      <slot name="badge" />
    </span>
  </div>
</template>

<script>
export default {
  name: 'VHBadge',
  props: {
    /**
     * Badge Color: white, blue, orange.
     */
    color: {
      type: String,
      default: 'blue',
    },
  },
  computed: {
    computedClasses() {
      return {
        'VHBadge--white': this.color === 'white',
        'VHBadge--blue': this.color === 'blue',
        'VHBadge--orange': this.color === 'orange',
      };
    },
  },
};
</script>

<style scoped>
.VHBadgeContainer {
  display: inline-block;
  position: relative;
}

.VHBadge {
  position: absolute;
  display: flex;
  top: 8px;
  right: 10px;
  height: 16px;
  min-width: 16px;
  padding: 0 2px;
  z-index: 1;
  align-items: center;
  justify-content: center;
  font-size: 8px;
  border-radius: 10px;
  transform: scale(1) translate(90%, -60%);
  transform-origin: 100% 0%;
}

.VHBadge--white {
  background-color: var(--color--white);
  color: var(--text-color);
}

.VHBadge--blue {
  background-color: var(--color--blue400);
  color: var(--color--white);
}

.VHBadge--orange {
  background-color: var(--color--orange500);
  color: var(--color--white);
}
</style>
