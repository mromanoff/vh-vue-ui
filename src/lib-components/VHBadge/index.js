import VHBadge from './VHBadge.vue';

export default VHBadge;
export { VHBadge };
