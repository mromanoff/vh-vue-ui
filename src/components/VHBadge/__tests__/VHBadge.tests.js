import { shallowMount } from '@vue/test-utils';
import VHBadge from '../VHBadge.vue';

describe('VHBadge.vue', () => {
  let wrapper;

  it('renders default VHBadge', () => {
    wrapper = shallowMount(VHBadge);
    expect(wrapper.props().color).toBe('blue');
    expect(wrapper).toMatchSnapshot();
  });

  it('renders white VHBadge', () => {
    wrapper = shallowMount(VHBadge, {
      propsData: {
        color: 'white',
      },
    });
    expect(wrapper.props().color).toBe('white');
    expect(wrapper).toMatchSnapshot();
  });
});
