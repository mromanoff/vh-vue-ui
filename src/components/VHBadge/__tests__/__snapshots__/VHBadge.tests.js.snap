// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VHBadge.vue renders default VHBadge 1`] = `<div class="VHBadgeContainer"><span></span> <span class="VHBadge VHBadge--blue"></span></div>`;

exports[`VHBadge.vue renders white VHBadge 1`] = `<div class="VHBadgeContainer"><span></span> <span class="VHBadge VHBadge--white"></span></div>`;
